### PR TITLE
fix(transport): cap stdio line length

### PR DESCRIPTION
**Pull request by SVilgelm**
### 📌 Summary

The stdio reader used `AsyncBufReadExt::lines()`, which buffers until a newline with no limit, so a peer streaming data without `\n` could force unbounded memory growth. Incoming lines are now capped; an over-long message is dropped (resynced to the next newline) and logged, keeping the stream alive.

### 🔍 Related Issues

- Related to #141

### ✨ Changes Made

- Replace `lines()` with a capped `fill_buf`/`consume` reader (4 MiB limit).
- Drop and log lines exceeding the cap instead of buffering them.
- Add unit tests for normal, trailing-newline-less, and oversized lines.

### CODE REVIEW

File: crates/rust-mcp-transport/src/mcp_stream.rs [hunk 1]
```diff
@@ -107,41 +110,47 @@ impl MCPStream {
...
+                            }
+                            Ok(LineRead::TooLong) => {
+                                // Drop the oversized message and keep the stream alive.
+                                tracing::warn!(
+                                    "dropping incoming message exceeding {MAX_LINE_LENGTH} bytes"
+                                );
+                                continue;
+                            }
+                            Ok(LineRead::Line(line)) => {
+                                tracing::trace!("raw payload: {}", line);
```
Comment (hashemix) [inline crates/rust-mcp-transport/src/mcp_stream.rs:135]: This can now log up to 4 MiB. For trace! level this is unlikely in production, but better to consider limiting:


```suggestion
                                tracing::trace!("raw payload: {}", &line[..line.len().min(1024)]);
```

File: crates/rust-mcp-transport/src/mcp_stream.rs [hunk 2]
```diff
@@ -150,3 +159,124 @@ impl MCPStream {
...
+
+    #[tokio::test]
+    async fn drops_oversized_line_and_resyncs() {
+        let mut data = vec![b'a'; 100];
+        data.push(b'\n');
+        data.extend_from_slice(b"ok\n");
+        let out = collect_lines(&data, 10).await;
+        assert_eq!(out, vec![Err("too-long"), Ok("ok".to_string())]);
+    }
+}
```
Comment (hashemix) [inline crates/rust-mcp-transport/src/mcp_stream.rs:282]: can we add two missing edge cases:

```rs
#[tokio::test]
async fn accepts_line_at_exact_max() {
    let data = format!("{}\n", "a".repeat(10)); // 10 bytes 'a' + newline
    let out = collect_lines(data.as_bytes(), 10).await;
    assert_eq!(out, vec![Ok("a".repeat(10))]);
}

#[tokio::test]
async fn resyncs_after_consecutive_oversized_lines() {
    let mut data = vec![b'a'; 100];
    data.push(b'\n');
    data.extend_from_slice(b"too-big-again\nok\n");
    let out = collect_lines(&data, 10).await;
    assert_eq!(out, vec![Err("too-long"), Err("too-long"), Ok("ok".to_string())]);
}
```

**Review by hashemix — CHANGES_REQUESTED** [review]

**Comment by hashemix** [comment]
Closing this in favor of #164
